src/connection/socket.ts

```diff
@@ -91,13 +91,13 @@ export class RethinkDBSocket extends EventEmitter {
         type: RethinkDBErrorType.CONNECTION,
       });
     }
-    const { tls = false, ...options } = this.connectionOptions;
+    const options = this.connectionOptions;
```

Comment:
I see no good reason of doing this because the `tls` field is not an option for TLS or standard socket connection, it just helps us to differ if the connection is secure or not.

Reply:
The problem here is that it stops typescript from being able to infer the type of `options` based on the value of `tls`, requiring an explicit cast to check the value of `rejectUnauthorized`.